### ISSUE TITLE
API call to extend task end time

### ISSUE THREAD

**Issue by arlake228** [perfsonar/pscheduler#88]
It would be useful to have a call that allows you to extend the end time of the task. The main use case is that in perfsonar an individual node has a bunch of regular tasks it wants to be executed. It will submit some of those to remote pschedulers (e.g. tests starting at the remote node and coming back to the local node). If the local node goes away (e.g. the host is decommissioned) the task will still remain on the remote server until the end time (or forever if there isn't one). 

In addition to this, it's also desirable to have uninterrupted testing while everything is still up. Without the ability to extend the end time for things like tests created via the GUI or the MeshConfig, it has to decide whether to forgo setting end time and risk a task getting orphaned forever,set an end time and risk interrupting tests or having duplicate overlapping test definitions for windows of time. While the latter two aren't terrible, it would be ideal if we could set an end time reasonably far in the future when creating a task and periodically extend it, essentially acting as a renewal of the task. 

Definitely not required for RC1, but  suspect it might be just a matter of having the a REST call update the end time field in the database, and our sliding time-horizon will magically do the rest of the work for us. That may be wishful thinking, so if its even more complicated we can push it out past 1.0 if need be.

**Comment by mfeit-internet2** [comment]
> If the local node goes away (e.g. the host is decommissioned) the task will still remain on the remote server until the end time (or forever if there isn't one). 

I think the way to solve this will be to purge non-lead tasks that haven't seen any activity in some time.  We could either do it by a fixed period or have the lead periodically retrieve the task through the REST API and have that be an indicator that there's interest somewhere.

> it would be ideal if we could set an end time reasonably far in the future when creating a task and periodically extend it, essentially acting as a renewal of the task. 

Any reason not to simply leave the end time off entirely and let it repeat forever or until canceled?

**Comment by mfeit-internet2** [comment]
Closed after discussion.  Having meshconfig recreate ongoing tasks is safer because if the second participant loses the task, there's nothing to put it back.  The expense of creating a new task isn't very big.

We _might_ revisit this after #148.
